### PR TITLE
[ADD] stock_dropshipping/sale_purchase: Migration scripts

### PR DESCRIPTION
**Pull request by pedrobaeza**
New module `sale_purchase` is a dependency of `stock_dropshipping` and part of the logic/fields is moved to this one, so both are handled together.

cc @Tecnativa

### CODE REVIEW

File: addons/stock_dropshipping/migrations/12.0.1.0/pre-migration.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.rename_xmlids(env.cr, [
+        ('stock_dropshipping.procurement_rule_drop_shipping',
+         'stock_rule_drop_shipping'),
```
Comment (MiquelRForgeFlow) [inline addons/stock_dropshipping/migrations/12.0.1.0/pre-migration.py]: needs the prefix

Comment (pedrobaeza) [reply addons/stock_dropshipping/migrations/12.0.1.0/pre-migration.py]: Fixed

**Comment by pedrobaeza** [comment]
I'm seeing that some code is not going to be executed as `sale_purchase` is a new module, so no migration script is executed. I'll change code to other module (probably base), but tomorrow.

**Comment by pedrobaeza** [comment]
OK, I have removed the update of the field moved to other module, as v12 handles it and it's not a translatable field.

**Review by MiquelRForgeFlow — APPROVED** [review]

**Comment by pedrobaeza** [comment]
In this project may be more convenient to not use the bot if we have a green CI for not tempting the luck, and also for not having to wait a lot (as the CI is slow).

**Comment by pedrobaeza** [comment]
Any way, thanks for your review!

**Pull request merged**